src/Connection/Internal/Http2ConnectionProcessor.php

```diff
@@ -1035,19 +1017,18 @@ public function request(Request $request, Cancellation $cancellation, Stream $st
                     $writeFuture->await($cancellation);
 
                     if ($chunk === null) {
-                        $http2stream->requestBodyCompletion->complete();
+                        $http2stream->requestBodyCompletion->complete(); // TODO Move after loop?
```

Comment:
I think moving this after loop would be better to be triggered at a similar time to the event listener.